src/components/PromiseList/PromiseList.svelte

```diff
@@ -3,9 +3,9 @@
 	import PromiseCard from './PromiseCard.svelte';
 	import Grid from 'carbon-icons-svelte/lib/Grid.svelte';
 	import List from 'carbon-icons-svelte/lib/List.svelte';
-	import type { PromiseSummary } from '$models/promise';
+	import type { ComponentProps } from 'svelte';
 
-	export let summaries: PromiseSummary[];
+	export let summaries: ComponentProps<PromiseCard>['promiseSummary'][];
```

Comment:
After you get rid of promiseSummary, you can use TypeScript `Omit` to exclude isList from ComponentProps here.